response_operations_ui/views/admin/manage_user_accounts.py

```diff
@@ -66,7 +72,171 @@ def manage_user_accounts():
+
+
+@admin_bp.route("/manage-account/<user_id>", methods=["POST"])
+@login_required
+def update_account_permissions(user_id):
+    _verify_user_admin_permission()
+
+    if user_id == session["user_id"]:
+        flash("You cannot modify your own user account", "error")
+        logger.error("User tried to modify their own account", user_id=user_id)
+        return redirect(url_for("admin_bp.manage_user_accounts"))
+
+    # Get user from uaa, so we have a fresh set of permissions to look at
+    user = get_user_by_id(user_id)
+    if user is None:
+        flash("User does not exist", "error")
+        return redirect(url_for("admin_bp.manage_user_accounts"))
+
+    try:
+        groups = get_groups()
+    except HTTPError:
+        flash("Failed to get groups, please try again", "error")
+        return redirect(url_for("admin_bp.manage_user_accounts"))
+
+    form = EditUserPermissionsForm(request.form)
+    user_groups = [group["display"] for group in user["groups"]]
+
+    translated_permissions = {
+        "surveys_edit": "surveys.edit",
+        "reporting_units_edit": "reportingunits.edit",
+        "respondents_edit": "respondents.edit",
+        "respondents_delete": "respondents.delete",
+        "messages_edit": "messages.edit",
+        "users_admin": "users.admin",
+    }
+
+    # Because we can't add or remove in a batch, if one of them fail then we can leave the user in a state that wasn't
+    # intended.  Though it can be easily fixed by trying again.
+
+    was_permission_changed = False
+    for permission, is_ticked in form.data.items():
+        # Translate the permission, so we have the uaa form of it
+        translated_permission = translated_permissions[permission]
+        group_details = next(item for item in groups["resources"] if item["displayName"] == translated_permission)
+        group_id = group_details["id"]
+        if is_ticked:
+            if translated_permission in user_groups:
+                continue  # Nothing to do, already part of the group
+            try:
+                add_group_membership(user_id, group_id)  # Ticked and not in group, need to add it
+            except HTTPError:
+                flash(f"Failed add [{permission}] to the user, please try again", "error")
+                return redirect(url_for("admin_bp.manage_account", user=user["emails"][0]["value"]))
+            was_permission_changed = True
+
+        if translated_permission in user_groups:
+            try:
+                remove_group_membership(user_id, group_id)  # Not ticked but in group, need to remove it
+            except HTTPError:
+                flash(f"Failed remove [{permission}] to the user, please try again", "error")
+                return redirect(url_for("admin_bp.manage_account", user=user["emails"][0]["value"]))
+            was_permission_changed = True
+        continue  # Nothing to do, already not in the group
```

Comment:
I disagree.  You're right about the continue not doing anything because it's the last line, the comment is useful as I'm trying to highlight the 4 states that a permission can be in and what we do in each case, those being:
- Box ticked and already in group
- Box ticked and not in group
- Box not ticked and already in group
- Box not ticked and not in group